rocketpy/motors/motor.py

```diff
@@ -1179,6 +1214,145 @@ def __init__(
+        coordinate_system_orientation : string, optional
+            Orientation of the motor's coordinate system. The coordinate system
+            is defined by the motor's axis of symmetry. The origin of the
+            coordinate system may be placed anywhere along such axis, such as
+            at the nozzle area, and must be kept the same for all other
+            positions specified. Options are "nozzle_to_combustion_chamber" and
+            "combustion_chamber_to_nozzle". Default is
+            "nozzle_to_combustion_chamber".
+        Returns
+        -------
+        Generic Motor object
+        """
+        if isinstance(file_name, str):
+            if file_name[-3:] == "eng":
+                _, description, thrust_source = Motor.import_eng(file_name)
+            else:
+                raise ValueError("File must be a .eng file.")
+        else:
+            raise ValueError("File name must be a string.")
+
+        thrust = Function(thrust_source, "Time (s)", "Thrust (N)", "linear", "zero")
+
+        # handle eng parameters
+
+        if chamber_radius:
+            if not isinstance(chamber_radius, (int, float)):
+                raise ValueError("Chamber radius must be a number.")
+        else:
+            chamber_radius = float(description[1])  # get motor diameter
+
+        if chamber_height:
+            if not isinstance(chamber_height, (int, float)):
+                raise ValueError("Chamber height must be a number.")
+        else:
+            chamber_height = float(description[2])  # get motor length
+
+        if propellant_initial_mass:
+            if not isinstance(propellant_initial_mass, (int, float)):
+                raise ValueError("Propellant initial mass must be a number.")
+        else:
+            propellant_initial_mass = float(description[-3])
+
+        if dry_mass:
+            if not isinstance(dry_mass, (int, float)):
+                raise ValueError("Dry mass must be a number.")
+        else:
+            total_mass = float(description[-2])
+            dry_mass = total_mass - propellant_initial_mass
+
+        e_vel = thrust.integral(*(0.0, 4.897)) / propellant_initial_mass
+        print(f"exhaust velocity: {e_vel:.2f} m/s ")
+
+        gen_motor = GenericMotor(
+            thrust_source=thrust,
+            burn_time=burn_time,
+            chamber_radius=chamber_radius,
+            chamber_height=chamber_height,
+            chamber_position=chamber_position,
+            propellant_initial_mass=propellant_initial_mass,
+            nozzle_radius=nozzle_radius,
+            dry_mass=dry_mass,
+            center_of_dry_mass_position=center_of_dry_mass_position,
+            dry_inertia=dry_inertia,
+            nozzle_position=nozzle_position,
+            reshape_thrust_curve=reshape_thrust_curve,
+            interpolation_method=interpolation_method,
+            coordinate_system_orientation=coordinate_system_orientation,
+        )
+        return gen_motor
```

Comment:
Save lines, save time.

```suggestion
        return GenericMotor(
            thrust_source=thrust,
            burn_time=burn_time,
            chamber_radius=chamber_radius,
            chamber_height=chamber_height,
            chamber_position=chamber_position,
            propellant_initial_mass=propellant_initial_mass,
            nozzle_radius=nozzle_radius,
            dry_mass=dry_mass,
            center_of_dry_mass_position=center_of_dry_mass_position,
            dry_inertia=dry_inertia,
            nozzle_position=nozzle_position,
            reshape_thrust_curve=reshape_thrust_curve,
            interpolation_method=interpolation_method,
            coordinate_system_orientation=coordinate_system_orientation,
        )
```